wicket-core/src/main/java/org/apache/wicket/resource/FileSystemResourceReference.java

```diff
@@ -0,0 +1,181 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.wicket.resource;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.wicket.request.resource.IResource;
+import org.apache.wicket.request.resource.ResourceReference;
+import org.apache.wicket.util.lang.Args;
+
+/**
+ * This resource reference is used to provide a reference to a resource based on Java NIO FileSystem
+ * API.<br>
+ * <br>
+ * To implement a mime type detection refer to the documentation of
+ * {@link java.nio.file.Files#probeContentType(Path)} and provide an implementation for
+ * java.nio.file.spi.FileTypeDetector in the META-INF/services folder for jars or in the
+ * /WEB-INF/classes/META-INF/services folder for webapps<br>
+ * <br>
+ * You can optionally override {@link #getMimeType()} to provide an inline mime type detection,
+ * which is preferred to the default detection.<br>
+ * <br>
+ * Example:
+ * 
+ * <pre>
+ * <code>
+ * Path path = FileSystemResourceReference.getPath(URI.create("jar:file:///folder/file.zip!/folderInZip/video.mp4"));
+ * add(new Video("video", new FileSystemResourceReference(path)));
+ * </code>
+ * </pre>
+ * 
+ * @author Tobias Soloschenko
+ */
+public class FileSystemResourceReference extends ResourceReference
+{
+	private static final long serialVersionUID = 1L;
+
+	private final Path path;
+
+	private static Map<URI, FileSystem> fileSystemURIs = new HashMap<URI, FileSystem>();
+
+	/**
+	 * Creates a file system resource reference based on the given path
+	 * 
+	 * @param name
+	 *            the name of the resource reference to expose data
+	 * @param path
+	 *            the path to create the resource reference
+	 */
+	public FileSystemResourceReference(String name, Path path)
+	{
+		super(Args.notNull(name, "name"));
+		Args.notNull(path, "path");
+		this.path = path;
+	}
+
+	/**
+	 * Creates a file system resource reference based on the given path
+	 * 
+	 * @param path
+	 *            the path to create the resource reference (the name is used to expose the data)
+	 */
+	public FileSystemResourceReference(Path path)
+	{
+		super(Args.notNull(path, "path").getFileName().toString());
+		this.path = path;
+	}
+
+	/**
+	 * Creates a new {@link org.apache.wicket.markup.html.media.FileSystemResource} and applies the
+	 * path to it.
+	 */
+	@Override
+	public IResource getResource()
+	{
+		return getFileSystemResource();
+	}
+
+	/**
+	 * Override to apply a custom mime type without implementing a mime type detection
+	 * 
+	 * @return the mime type
+	 * @throws IOException
+	 *             if the mime type could'nt be read
+	 */
+	protected String getMimeType() throws IOException
```

Comment:
Yes you are right - originally I wanted the FSRR#getMimeType() to be used for the FSR if overridden. I will reimplement it.